ambry-router/src/test/java/com.github.ambry.router/GetManagerTest.java

```diff
@@ -133,6 +136,34 @@ public void testFailureOnAllPollThatSends()
       Assert.assertEquals("Exception received should be router closed error", RouterErrorCode.RouterClosed,
           routerException.getErrorCode());
     }
+
+    try {
+      router.getBlob(blobId).get();
+      Assert.fail("operation should have thrown");
+    } catch (ExecutionException e) {
+      RouterException routerException = (RouterException) e.getCause();
+      Assert.assertEquals("Exception received should be router closed error", RouterErrorCode.RouterClosed,
+          routerException.getErrorCode());
+    }
+  }
+
+  /**
+   * Do a getBlob on the given blob id and ensure that all the data is fetched and is correct.
+   * @param blobId the id of the blob (simple or composite) that needs to be fetched and compared.
+   * @throws Exception
+   */
+  private void getBlobAndCompareContent(String blobId)
+      throws Exception {
+    ByteBufferAsyncWritableChannel getChannel = new ByteBufferAsyncWritableChannel();
+    router.getBlob(blobId).get().readInto(getChannel, null);
+    int readBytes = 0;
+    do {
+      ByteBuffer buf = getChannel.getNextChunk();
+      while (buf.hasRemaining()) {
+        Assert.assertEquals("Get and Put blob content should match", putContent[readBytes++], buf.get());
+      }
+      getChannel.resolveOldestChunk(null);
+    } while (readBytes < putContent.length);
   }
```

Comment:
I think that there are two final checks required to ensure that there is no "extra" data (in the same order).

1. The future returned from the `readInto()` should be "done".
2. There should be no more chunks to read (`assertNull("There was unexpected extra data", getChannel.getNextChunk(0))`).